astropy/nddata/tests/test_ccddata.py

```diff
@@ -55,7 +56,7 @@ def test_ccddata_unit_cannot_be_set_to_none():
 def test_ccddata_meta_header_conflict():
     with pytest.raises(ValueError) as exc:
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=r"can't have both header and meta."):
```
Good catch! But generally we recommend the more convenient `match=` functionality for new tests now
(https://docs.astropy.org/en/latest/development/testguide.html#testing-exceptions); won't need the `as exc` then anymore – the same applies to all the added tests below.

Reply:
@parejkoj -- let me know if you want me to make the test changes.

Reply:
@mwcraig yes, if you could make those test changes, that would be useful, since I'm less familiar with pytest.

Reply:
@dhomeier @parejkoj -- I'm going to open a separate PR in a few minutes to fix all of the nddata tests to use `match`, including the one you suggested, Derek.

I'll revert the change @parejkoj  made at this spot since it is unrelated to the psf functionality.

Reply:
Edit: I won't bother with the revert, since any conflict can be fixed during the rebase.